conda/petsc/build.sh

```diff
@@ -1,119 +1,164 @@
-export FFLAGS=$(echo ${FFLAGS:-} | sed -E 's@\-fdebug\-prefix\-map[^ ]*@@g')
-export FCFLAGS="$FFLAGS"
-export HYDRA_LAUNCHER=fork
-
-if [[ $(uname) == Darwin ]]; then
-    if [[ $HOST == arm64-apple-darwin20.0.0 ]]; then
-        CFLAGS="${CFLAGS} -mcpu=apple-a12"
-        CXXFLAGS="${CXXFLAGS} -mcpu=apple-a12"
-        FFLAGS="${FFLAGS} -march=armv8.3-a"
-        FCFLAGS="${FCFLAGS} -march=armv8.3-a"
-    else
-        CFLAGS="${CFLAGS} -march=core2 -mtune=haswell"
-        CXXFLAGS="${CXXFLAGS} -march=core2 -mtune=haswell"
-        FFLAGS="${FFLAGS} -I$PREFIX/include"
-        FCFLAGS="${FCFLAGS} -I$PREFIX/include"
-    fi
-else
-    CFLAGS="${CFLAGS} -march=nocona -mtune=haswell"
-    CXXFLAGS="${CXXFLAGS} -march=nocona -mtune=haswell"
-    FFLAGS="${FFLAGS} -I$PREFIX/include"
-    FCFLAGS="${FCFLAGS} -I$PREFIX/include"
-fi
-
-# Remove std=C++17 from CXXFLAGS as we specify the C++ dialect for PETSc as C++17 in configure_petsc.
-# Specifying both causes an error as of PETSc 3.17.
+#
+#export CC=$(basename "$CC")
+#export CXX=$(basename "$CXX")
+#export FC=$(basename "$FC")
+#
+## feed-stock recommendation
+## scrub debug-prefix-map args, which cause problems in pkg-config
+#export CFLAGS=$(echo ${CFLAGS:-} | sed -E 's@\-fdebug\-prefix\-map[^ ]*@@g')
+#export CXXFLAGS=$(echo ${CXXFLAGS:-} | sed -E 's@\-fdebug\-prefix\-map[^ ]*@@g')
+#export FFLAGS=$(echo ${FFLAGS:-} | sed -E 's@\-fdebug\-prefix\-map[^ ]*@@g')
+#export FCFLAGS="$FFLAGS"
+#export HYDRA_LAUNCHER=fork
+#
+#if [[ $(uname) == Darwin ]]; then
+#    BUILD_VARIANT=""
+#    if [[ $HOST == arm64-apple-darwin20.0.0 ]]; then
+#        CFLAGS="${CFLAGS} -mcpu=apple-a12"
+#        CXXFLAGS="${CXXFLAGS} -mcpu=apple-a12"
+#        FFLAGS="${FFLAGS} -march=armv8.3-a"
+#        FCFLAGS="${FCFLAGS} -march=armv8.3-a"
+#    else
+#        CFLAGS="${CFLAGS} -march=core2 -mtune=haswell"
+#        CXXFLAGS="${CXXFLAGS} -march=core2 -mtune=haswell"
+#        FFLAGS="${FFLAGS} -I$PREFIX/include"
+#        FCFLAGS="${FCFLAGS} -I$PREFIX/include"
+#    fi
+#else
+#    BUILD_VARIANT=${build_variant}
+#    CFLAGS="${CFLAGS} -march=nocona -mtune=haswell"
+#    CXXFLAGS="${CXXFLAGS} -march=nocona -mtune=haswell"
+#    FFLAGS="${FFLAGS} -I$PREFIX/include"
+#    FCFLAGS="${FCFLAGS} -I$PREFIX/include"
+#fi
+#
+## Remove std=C++17 from CXXFLAGS as we specify the C++ dialect for PETSc as C++17 in configure_petsc.
+## Specifying both causes an error as of PETSc 3.17.
 CXXFLAGS=${CXXFLAGS//-std=c++[0-9][0-9]}
+#
+## Stole this from petsc-feedstock in an attempt to solve the openmpi error:
+## mca_base_component_repository_open: unable to open mca_btl_openib: librdmacm.so.1: cannot open shared object file: No such file or directory
+#if [[ $mpi == "openmpi" ]]; then
+#  export LIBS="-Wl,-rpath,$PREFIX/lib -lmpi_mpifh -lgfortran"
+#elif [[ $mpi == "mpich" ]]; then
+#  export LIBS="-lmpifort -lgfortran"
+#fi
+#
+## Handle switches created by Conda variants
+ADDITIONAL_ARGS=""
```

Comment:
Where is this getting defined? You've commented out line 63. Is this (and its presence in the PETSC config line below) still necessary?

Reply:
I think I need to leave it, as I believe it is not possible to comment out a line continuation later:

```bash
configure_petsc \
      --COPTFLAGS=-O3 \
      --CXXOPTFLAGS=-O3 \
      --FOPTFLAGS=-O3 \
      --with-x=0 \
      --with-ssl=0 \
      --with-hdf5-dir=$HDF5_DIR \
      --with-mpi-dir=$PREFIX \
      AR="$AR" \
      RANLIB="$RANLIB" \
      CFLAGS="$CFLAGS" \
      CXXFLAGS="$CXXFLAGS" \
      CPPFLAGS="$CPPFLAGS" \
      FFLAGS="$FFLAGS" \
      FCFLAGS="$FFLAGS" \
      LDFLAGS="$LDFLAGS" \
      ${ADDITIONAL_ARGS} \         <----------------------------------- this guy
      --prefix=$PREFIX/petsc || (tail -400 configure.log && exit 1)
```
I could just remove both until the time is needed (this will be needed for pytorch stuff soon (tm).

Reply:
It's fine to leave it in if you leave a good TODO comment about why it's there and empty.